lib/record_store/zone/config/template.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module RecordStore
+  class Zone
+    class Config
+      class TemplateContext
+        def self.build(record:, all_records:)
+          new(record: record, all_records: all_records)
+        end
+
+        def initialize(record:, all_records:)
+          @record = record
+          @all_records = all_records
+        end
+
+        def get_binding
+          binding
+        end
+
+        private
+
+        attr_reader :record, :all_records
+      end
+
+      class Template
+        class Error < StandardError; end
+        class InvalidFilepathError < Error; end
+
+        class << self
+          def from_file(filename:)
+            filepath = template_filepath_for(filename: filename)
+
+            raise InvalidFilepathError, "Could not find template '#{filename}' at location #{filepath}\n"\
+              unless File.exists?(filepath)
+
+            template_file = File.read(filepath)
+            filters_for_records_to_template = YAML.load(template_file).deep_symbolize_keys[:each_record]
```

Comment:
Could the exception be done as a `rescue` from a `read` error ? This would give the exact error, including if there's an issue reading the file that might exist but not have the right permissions, for instance

We might also want to rescue on the `YAML.load` error to handle the case where the file is not properly formatted in YAML.

Reply:
yeah we can use the base exception if we want, this just reformatted the info a bit. 

Will think about how to handle the yaml errors as well.

Reply:
in other places we dont handle these sorts of errors unless we want to add on something special to the reporting, or have error handling logic, which we dont in these cases, so i'm opting to just let errors bubble up naturally for filepath and yaml format problems

Reply:
It totally makes sense! Thanks for checking things to arrive to that conclusion :)